proton-j/src/main/java/org/apache/qpid/proton/engine/Event.java

```diff
@@ -80,15 +83,46 @@
         SELECTABLE_WRITABLE,
         SELECTABLE_EXPIRED,
         SELECTABLE_ERROR,
-        SELECTABLE_FINAL
+        SELECTABLE_FINAL,
+        
+        /**
+         * This value must never be used to generate an event, it's only used as
+         * a guard when casting custom EventTypes to builtin {@link Type} via
+         * {@link Event#getBuiltinType()}.
+         */
+        NOT_A_BUILTIN_TYPE;
```

Comment:
I'll go with something on the theme NON_CORE_EVENT?

The getBuiltinType() is from a previous iteration of the patch and should have been renamed to getType()